### PR TITLE
fix(compile): .git doesn't exist on nixos

### PR DESCRIPTION
**Pull request by nullchilly**
cc @stasjok #352

I will also look into the re-compile issue, sorry..

### CODE REVIEW

File: lua/catppuccin/init.lua [hunk 1]
```diff
@@ -120,9 +120,9 @@ function M.setup(user_conf)
 	end
 
 	local config_path = debug.getinfo(2).source:sub(2) -- Get user config path
-	local git_path = debug.getinfo(1).source:sub(2, -24) .. ".git" .. M.path_sep .. "ORIG_HEAD" -- Get git path
+	local git_path = debug.getinfo(1).source:sub(2, -24) -- Get git path
 	local config = vim.fn.getftime(config_path) -- getftime is 2 time faster than fs_stat, for benchmark see #352
-	local git = vim.fn.getftime(git_path) -- Parsed config & git stat
+	local git = vim.fn.getftime(git_path .. ".git" .. M.path_sep .. "ORIG_HEAD") -- Parsed config & git stat
 
 	local cur_date = (config == 1 and config_path or config) .. (git == 1 and git_path or git) -- nix mtime is always 1 so cache path instead
```
Comment (stasjok) [inline lua/catppuccin/init.lua:127]:
```suggestion
	local cur_date = (config == 1 and config_path or config) .. (git == -1 and git_path or git) -- nix mtime is always 1 so cache path instead
```

**Comment by nullchilly** [comment]
@stasjok I think I should just disable the fragile user config mtime to be honest

**Comment by nullchilly** [comment]
Because it would need three seperate files cache at this point

`date`, `git` and `config`

**Comment by stasjok** [comment]
Latest version has updated `date.txt` with:
```
1669531780/nix/store/xi9ym8i2zlxhm41pdqdcmxx1ynh5b359-vim-pack-dir/pack/myNeovimPackages/start/catppuccin-nvim/.git/ORIG_HEAD
```
So it should work. `*_compiled.lua` files aren't updated, but it's a different problem not related to nix.

**Comment by nullchilly** [comment]
Alright, I will merge this first, then open another PR to fix that different problem

**Pull request merged**